webpack/assets/javascripts/react_app/components/Bookmarks/components/BookmarkForm/index.js

```diff
@@ -1,15 +1,23 @@
 import { connect } from 'react-redux';
-
 import BookmarkForm from './BookmarkForm';
 import { submitForm } from '../../../../redux/actions/common/forms';
+import { selectAutocompleteSearchQuery } from '../../../AutoComplete/AutoCompleteSelectors';
+
+const mapStateToProps = (state, ownProps) => ({
+  initialValues: {
+    public: true,
+    query:
+      selectAutocompleteSearchQuery(
+        { autocomplete: { id: 'searchBar' }, ...state },
+        'searchBar',
+        ownProps
+      ) || '',
```

Comment:
imo we should use it differently:
`selectAutocompleteSearchQuery(state, 'searchBar', ownProps) || '',`
or `selectAutocompleteSearchQuery(state, 'searchBar', { controller ) || '',`

and we might want to set it in a `BookmarkFormSelectors.js` file
and consume it here in a shorter version, e.g: `query: selectQuery(state, controller)`

Reply:
@laviro see [this comment](https://github.com/theforeman/foreman/pull/7265#discussion_r368745668) for the reason I did it that way.  I'll see if I can come up with another solution.. :confused:

Reply:
After looking some more at this I don't see why the `AutoComplete` search query even has to be in Redux at all.  However at this point we are way outside the scope of this PR and I would prefer to do a follow-up PR for additional refactoring.

Reply:
Can you open a redmine for this issue? :)

Reply:
@MariaAga Ok, opened https://projects.theforeman.org/issues/28914